### PR TITLE
Release 0.9.18

### PR DESCRIPTION
**Pull request by yannikschaelte**
* Add create_sqlite_db_id convenience function to create database names.
* Temporarily require cffi=1.12.2 for rpy2 on travis (all #185).
* Introduce UniformAcceptor and SimpleFunctionAcceptor classes to streamline
  the traditional acceptance step.
* Add AcceptorResult and allow weights in the acceptance step (all #184).

### CODE REVIEW

File: pyabc/acceptor/acceptor.py [hunk 1]
```diff
@@ -0,0 +1,288 @@
+"""
+Acceptor
+--------
+
+After summary statistics for samples for given parameters have
```
Comment (elbaraim) [inline pyabc/acceptor/acceptor.py]: After summary statistics for samples for given parameters have --> After summary statistics for samples **of** given parameters have ?

File: pyabc/acceptor/acceptor.py [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+    can frequently occur when continuing a stopped run. A different behavior
+    is easy to implement.
+    """
+    # first test current criterion, which is most likely to fail
+    d = distance_function(x, x_0, t, par)
+    accept = d <= eps(t)
+
+    if accept:
+        # also check against all previous distances and acceptance criteria
+        for t_prev in range(0, t):
```
Comment (elbaraim) [inline pyabc/acceptor/acceptor.py]: can be replaced with `range(t)`

File: pyabc/acceptor/acceptor.py [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+                # ignore as of now
+                accept = True
+
+    return AcceptorResult(distance=d, accept=accept)
+
+
+class UniformAcceptor(Acceptor):
+    """
+    Base acceptance on the distance function and a uniform error distribution
+    between -eps and eps. This is the most common acceptance criterion in ABC.
```
Comment (elbaraim) [inline pyabc/acceptor/acceptor.py]: for clarity:  -eps and **+**eps ?

File: pyabc/model.py [hunk 4]
```diff
@@ -19,10 +19,12 @@ class ModelResult:
     distances and accepted/rejected.
     """
 
-    def __init__(self, sum_stats=None, distance=None, accepted=None):
+    def __init__(self,
+                 sum_stats=None, distance=None, accepted=None, weight=1.0):
```
Comment (elbaraim) [inline pyabc/model.py]: break the arguments in different text lines ?

File: pyabc/acceptor/acceptor.py (hunk 2 above)
Comment (yannikschaelte) [reply pyabc/acceptor/acceptor.py]: True, but I like to write things explicitly :)

File: pyabc/model.py (hunk 4 above)
Comment (yannikschaelte) [reply pyabc/model.py]: will do and also add type hints, thx

**Pull request merged**